axelor-sale/src/main/java/com/axelor/apps/sale/service/loyalty/LoyaltyAccountPointsManagementServiceImpl.java

```diff
@@ -0,0 +1,64 @@
+package com.axelor.apps.sale.service.loyalty;
+
+import com.axelor.apps.base.db.Company;
+import com.axelor.apps.base.db.Partner;
+import com.axelor.apps.sale.db.LoyaltyAccount;
+import com.axelor.apps.sale.db.repo.LoyaltyAccountRepository;
+import com.google.inject.Inject;
+import com.google.inject.persist.Transactional;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+
+public class LoyaltyAccountPointsManagementServiceImpl
+    implements LoyaltyAccountPointsManagementService {
+
+  protected LoyaltyAccountRepository loyaltyAccountRepository;
+  protected LoyaltyAccountService loyaltyAccountService;
+
+  @Inject
+  public LoyaltyAccountPointsManagementServiceImpl(
+      LoyaltyAccountRepository loyaltyAccountRepository,
+      LoyaltyAccountService loyaltyAccountService) {
+    this.loyaltyAccountRepository = loyaltyAccountRepository;
+    this.loyaltyAccountService = loyaltyAccountService;
+  }
+
+  @Override
+  public void incrementLoyaltyPointsFromAmount(
+      Partner partner, Company company, BigDecimal amount) {
+    LoyaltyAccount loyaltyAccount = loyaltyAccountService.getLoyaltyAccount(partner, company);
+    BigDecimal earnedPoints = pointsEarningComputation(amount);
+    incrementPoints(loyaltyAccount, earnedPoints);
+  }
+
+  @Transactional(rollbackOn = {Exception.class})
+  public void updatePoints(LoyaltyAccount loyaltyAccount, BigDecimal points, boolean isIncrement) {
```

Comment:
isIncrement is not necessary, either points is > 0 or <= 0, BigDecimal.add should do the job